### ISSUE TITLE
Train for custom data

### ISSUE THREAD

**Issue by mohitjain1943** [tuanphan09/captcha-recognition#9]
Hi @tuanphan09 ,

Can we use these code to train custom data instead of captcha?
For Images like

![INCOMETAXDEPARTMENT](https://user-images.githubusercontent.com/6769471/81141842-6ff86f80-8f8b-11ea-829d-1593d5b64aaf.png)
![PanCard15885101387](https://user-images.githubusercontent.com/6769471/81141855-7850aa80-8f8b-11ea-8237-808aa5d89fa9.png)
![PanCard15885101390](https://user-images.githubusercontent.com/6769471/81141857-78e94100-8f8b-11ea-8d5c-e6dd1ef81143.png)

I am trying to train with my custom images getting below error.

![Capture](https://user-images.githubusercontent.com/6769471/81142234-ae425e80-8f8c-11ea-9c68-20b81b96958e.PNG)

**Comment by tuanphan09** [comment]
Yes, you can use this repo to train your custom data.

I think you had error because some images cannot read by opencv2 or images shape is not right.
I just update my code with a little changes. You should pull the newest code to see where the error is.

**Comment by mohitjain1943** [comment]
Do we need all image in same width and height. becoz i am getting below error,

`Epoch 1/30
{} image shape is not the same as input test15885101604.png
{} image shape is not the same as input test15885101651.png
C:\Users\mohit\AppData\Local\Programs\Python\Python36\lib\site-packages\keras\utils\data_utils.py:616: UserWarning: The input 0 could not be retrieved. It could be because a worker has died.
  UserWarning)
{} image shape is not the same as input test15885101604.png`
 
I changed one line in model.py for input shape but still not working any suggestion. @tuanphan09 

this 
`inputShape = Input((width, height, 1))  # base on Tensorflow backend`
to this
`inputShape = Input(shape=(None, None, 1))  # base on Tensorflow backend`

based on this stackoverflow answer https://stackoverflow.com/questions/39814777/can-keras-deal-with-input-images-with-different-size

**Comment by tuanphan09** [comment]
Yes, you need all image in same width and height. If your images have not the same size, you have to resize it.

Then change the value of ``height, width`` in ``config.py``.

**Comment by mohitjain1943** [comment]
Where should i put resize code in `training.py` or In `model.py` ??

I resize all image using separate code after that i stated training i am getting below error
![Capture](https://user-images.githubusercontent.com/6769471/81256711-ef02ac00-904e-11ea-97d8-91e3a58e82fa.PNG)

I have below changes in config.py
`width = 200`
`height = 30`
`label_len = 40`

`characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ/-0123456789. abcdefghijklmnopqrstuvwxyz' `

**Comment by mohitjain1943** [comment]
It Worked after i changed batch_size to 8 in config.py

https://stackoverflow.com/questions/56847576/keyerror-val-loss-when-training-model
